p2p/simulations/http_test.go

```diff
@@ -565,6 +581,22 @@ func TestHTTPNodeRPC(t *testing.T) {
 	}
 	defer sub.Unsubscribe()
 
+	// make sure the subscription becomes active
+	var numActiveSubscriptions int64
+	for i := 0; i < 3; i++ {
```

Comment:
I think you should do a `for` and `select` loop with a timeout here. You're basically saying "timeout after 300 milliseconds" here, but in a more complicated way.